src/ApiGatewayWebSocket.js

```diff
@@ -0,0 +1,354 @@
+          }
+          else actionName = null;
+        }
+        if (typeof actionName !== 'string') actionName = null;
+        const action = actionName || '$default';
+        debugLog(`action:${action} on connection=${connection.connectionId}`);
+        const event = wsHelpers.createEvent(action, 'MESSAGE', connection, request.payload, this.options);
+
+        doAction(ws, connection.connectionId, action, event, true);
+
+        return h.response().code(204);
+      },
+    });
+
+    this.wsServer.route({
+      method: 'GET',
+      path: '/{path*}',
+      handler: (request, h) => h.response().code(426),
+    });
+
+    this.wsServer.route({
+      method: 'POST',
+      path: '/@connections/{connectionId}',
+      config: { payload: { parse: false } },
+      handler: (request, h) => {
+        debugLog(`got POST to ${request.url}`);
+        const getByConnectionId = (map, searchValue) => {
+          for (const [key, connection] of map.entries()) {
+            if (connection.connectionId === searchValue) return key;
+          }
+
+          return undefined;
+        };
+
+        const ws = getByConnectionId(this.clients, request.params.connectionId);
+        if (!ws) return h.response().code(410);
+        if (!request.payload) return '';
+        ws.send(request.payload.toString());
+        // console.log(`sent "${request.payload.toString().substring}" to ${request.params.connectionId}`);
+        debugLog(`sent data to connection:${request.params.connectionId}`);
+
+        return '';
+      },
+    });
+  }
+
+  _createWsAction(fun, funName, servicePath, funOptions, event) {
+    let handler; // The lambda function
+    Object.assign(process.env, this.originalEnvironment);
+
+    try {
+      if (this.options.noEnvironment) {
+        // This evict errors in server when we use aws services like ssm
+        const baseEnvironment = {
+          AWS_REGION: 'dev',
+        };
+        if (!process.env.AWS_PROFILE) {
+          baseEnvironment.AWS_ACCESS_KEY_ID = 'dev';
+          baseEnvironment.AWS_SECRET_ACCESS_KEY = 'dev';
+        }
+
+        process.env = Object.assign(baseEnvironment, process.env);
+      }
+      else {
+        Object.assign(
+          process.env,
+          { AWS_REGION: this.service.provider.region },
+          this.service.provider.environment,
+          this.service.functions[funName].environment
+        );
+      }
+      process.env._HANDLER = fun.handler;
+      handler = functionHelper.createHandler(funOptions, this.options);
+    }
+    catch (err) {
+      return this.serverlessLog(`Error while loading ${funName}`, err);
+    }
+
+    const actionName = event.websocket.route;
+    const action = { funName, fun, funOptions, servicePath, handler };
+
+    this.wsActions[actionName] = action;
+    this.serverlessLog(`Action '${event.websocket.route}'`);
+  }
+
+  _extractAuthFunctionName(endpoint) {
+    const result = authFunctionNameExtractor(endpoint, this.serverlessLog);
```

Comment:
It says it the documentation that authentication is not supported. Is it something else?